platform/linux-generic/odp_classification.c

```diff
@@ -738,6 +746,18 @@ odp_pmr_t odp_cls_pmr_create(const odp_pmr_param_t *terms, int num_terms,
 	return id;
 }
 
+odp_pmr_t odp_cls_pmr_create_opt(const odp_pmr_create_opt_t *opt,
+				 odp_cos_t src_cos, odp_cos_t dst_cos)
+{
+	(void)opt;
+	(void)src_cos;
+	(void)dst_cos;
+
+	ODP_UNIMPLEMENTED();
```

Comment:
This could just call odp_cls_pmr_create() and then the API implementation would be complete (although a bit limited in the allowed mark range).

Reply:
Are we on same page wrt implemetion?
`odp_pmr_t odp_cls_pmr_create_opt(const odp_pmr_create_opt_t *opt,
                                 odp_cos_t src_cos, odp_cos_t dst_cos)
{

        if (opt == NULL)
                return ODP_PMR_INVALID;

        if (opt->mark != 0) {
                ODP_UNIMPLEMENTED();
                return ODP_PMR_INVALID;
        }

        return odp_cls_pmr_create(opt->terms, opt->num_terms, src_cos, dst_cos);
}`

Reply:
There is no need to check that opt->mark is zero because it is required to be due to the max_mark capability value being zero. If you leave the check (I would just remove it), then I do not think ODP_UNIMPLEMENTED() is the right thing to do (since the function is implemented and does work). It is just the case that too large a mark value was passed, so returning ODP_PMR_INVALID would seem like the right thing to do.

Reply:
I thought of checking opt->mark against zero to catch rouge application which is not honoring the max_mark value.
In the current implementation the max_mark = 0, But In the future when we add support for the real mark, It needs to be compared against odp_cls_capability_t::max_mark. Is it adding up? Agree on ODP_UNIMPLEMENTED(), I can remove that

Reply:
Yes, that is fine.